include/timeline/widgets/FileItem.h

```diff
@@ -42,6 +42,7 @@ class FileItem : public QWidget
 
         FileItem(QSharedPointer<MatrixClient> client,
                  const QString &url,
+                 const QSharedPointer<QIODevice> data,
```

Comment:
Why is this needed? Unlike with image/audio/video, I'm not sure we can/should do much in the client with the data of an arbitrary opaque file.

Reply:
This is needed because [`TimelineView::addUserMessage`](https://github.com/mujx/nheko/pull/180/commits/5ccc484cea0f239c39a7e0f634b463048e059083#diff-f4908a06f6a57ca0c9989bd84b42b7c3R206) takes in a generic type `Widget`, which happens to be one of the four `FileItem`, `ImageItem`, `AudioItem`, and `VideoItem`. Therefore, we need those types to have the same interface (parameters to the constructor).

Reply:
I'm not particularly happy about having never-used arguments making things confusing on both ends for the sake of convenience to a template. That said, refactoring `addUserMessage` is probably out of scope for this PR.